File: introduction/ITP1_7_B-How many ways?.cs
using System;
using System.Linq;

public class Program {
  public static void Main() {
    string line;
    while((line = Console.ReadLine()) != "0 0") {
      int[] arr = line.Split().Select(int.Parse).ToArray();
      int n = arr[0], x = arr[1], c = 0;
      for(int i = 1; i < x/3; i++) c += Math.Max(Math.Min(n, (x-i+1)/2) - Math.Max(i+1, x-i-n), 0);
      Console.WriteLine(c);
    }
  }
}

File: introduction/ITP1_7_B-How many ways?.js
process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data', function(chunk) {
  var lines = chunk.split('\n');
  lines.some(function(line) {
    var arr = line.split(' ').map(Number);
    var n = arr[0], x = arr[1], c = 0;
    if(n+x == 0) return true;
    for(var i = 1; i < x/3; i++) c += parseInt(Math.max(Math.min(n, (x-i+1)/2) - Math.max(i+1, x-i-n), 0));
    console.log(c);
  });
});
